app/save-and-restore/ui/src/main/java/org/phoebus/applications/saveandrestore/ui/snapshot/SnapshotController.java

```diff
@@ -533,6 +539,92 @@ public void loadSaveSet(Node node){
+
+        List<ConfigPv> saveSetEntries = new ArrayList<ConfigPv>();
+
+        saveAndRestoreService.findNode(saveSetName, false);
+
+        //CREATE FOLDER
+        Node root = saveAndRestoreService.getRootNode();
+        Node newFolder = Node.builder()
+                .nodeType(NodeType.FOLDER)
+                .name(saveSetName)
+                .build();
+        Node newFolderNode;
+
+        try {
+            newFolderNode = saveAndRestoreService.createNode(root.getUniqueId(), newFolder);
+
+            //CREATE CONFIGURATION
+            Node newSateSetNode = Node.builder()
+                    .nodeType(NodeType.CONFIGURATION)
+                    .name(saveSetName)
+                    .build();
+            Node newTreeNode = saveAndRestoreService.createNode(newFolderNode.getUniqueId(), newSateSetNode);
+
+            for (String pv : pvList) {
+                ConfigPv configPv = ConfigPv.builder()
+                        .pvName(pv)
+                        .readOnly(false)
+                        .build();
+                saveSetEntries.add(configPv);
+            }
+
+            newTreeNode.putProperty(DESCRIPTION_PROPERTY, comment);
+            newTreeNode = saveAndRestoreService.updateSaveSet(newTreeNode, saveSetEntries);
+            List<ConfigPv> configPvs = saveAndRestoreService.getConfigPvs(newTreeNode.getUniqueId());
+            Node config = saveAndRestoreService.getNode(newTreeNode.getUniqueId());
+            List<SnapshotItem> snapshotItems = new ArrayList<>();
+            List<PVReader<VType>> readers = new ArrayList<>();
+
+            CountDownLatch countdownLatch = new CountDownLatch(pvList.size());
+            try {
+                for (ConfigPv pv : configPvs) {
+                    PVReader<VType> readerPv = GPClient.readAndWrite(patchPvName(pv.getPvName()))
+                            .addReadListener((event, p) -> {
+                                if (event.getType().contains(PVEvent.Type.VALUE)) {
+                                    SnapshotItem item = SnapshotItem.builder()
+                                            .value(p.getValue())
+                                            .readbackValue(null)
+                                            .configPv(pv)
+                                            .build();
+                                    snapshotItems.add(item);
+
+                                    countdownLatch.countDown();
+                                }
+                            })
+                            .start();
+                    readers.add(readerPv);
+                }
+
+                try {
+                    await().atMost(2, SECONDS).until(() -> readers.stream().allMatch(x -> x.isConnected()));
+                } catch (ConditionTimeoutException ex) {
+                    LOGGER.log(Level.WARNING, "Some PVs could not be saved because they are disconnected");
+                }
+
+                countdownLatch.await(1, SECONDS);
```

Comment:
Any fixed timeouts like atMost(2, SECONDS), 1 second, .. are likely wrong.
Also, CountDownLatch already offers `await(2, TimeUnit.SECOND)`. Do you really need to drag in another dependency to duplicate standard Java API?

Reply:
Is there a better way to determine the timeout? It takes a while for the PVs to connect and for data to be received. A timeout is definitely needed, but to what value should it be set?

We did have an issue with the PVs not connecting, and the code quietly continuing with some PVs not being saved. I will check if increasing the timeout is enough and that this library can be removed.

Reply:
The point is exactly that there's not one good universal value for the timeout. It depends on your site. The EPICS `caget` command defaults to a 1 second timeout, but if you have a larger network with several gateways, a 10 second timeout might be more appropriate. A tool that connects to 10000 PVs might take 30 seconds until all PVs connect.

That's why `caget` can use timeouts that you pass on the command line. Likewise, your code could use a preference setting to set the timeout, or pass that in, as done with some of the Displays Builder's https://github.com/ControlSystemStudio/phoebus/blob/master/app/display/runtime/src/main/java/org/csstudio/display/builder/runtime/script/PVUtil.java